src/daemonStarter.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { join } from 'path';
+import { spawn } from 'child_process';
+import { ExitCode } from './local-ssh/common';
+import { ILogService } from './services/logService';
+
+const sleep = (sec: number) => new Promise(resolve => setTimeout(resolve, sec * 1000));
+
+export async function ensureDaemonStarted(logService: ILogService, retry = 10) {
+    if (retry < 0) {
+        return;
+    }
+    const localAppProcess = await tryStartDaemon(logService);
+    localAppProcess.once('exit', code => {
+        switch (code) {
+            case ExitCode.OK:
+            case ExitCode.ListenPortFailed:
+                logService.error('exit with code: ' + code);
+                return;
+        }
+        logService.error('unexpectedly exit with code: ' + code + ' attempt retry: ' + retry);
+        ensureDaemonStarted(logService, retry - 1);
+    });
+    await sleep(1);
+}
+
+export async function tryStartDaemon(logService: ILogService) {
```

Comment:
if it fails because it is not already running then we should log an error to GCP, we should have an analytics event for whether it is succesfully started